stake-pool/program/tests/redelegate.rs

```diff
@@ -0,0 +1,1129 @@
+        transaction::{Transaction, TransactionError},
+    },
+    spl_stake_pool::{
+        error::StakePoolError, find_ephemeral_stake_program_address,
+        find_transient_stake_program_address, id, instruction, MINIMUM_RESERVE_LAMPORTS,
+    },
+};
+
+async fn setup(
+    do_warp: bool,
+) -> (
+    ProgramTestContext,
+    Hash,
+    StakePoolAccounts,
+    ValidatorStakeAccount,
+    ValidatorStakeAccount,
+    u64,
+    u64,
+) {
+    let mut context = program_test().start_with_context().await;
+    let rent = context.banks_client.get_rent().await.unwrap();
+    let stake_rent = rent.minimum_balance(std::mem::size_of::<stake::state::StakeState>());
+    let current_minimum_delegation = stake_pool_get_minimum_delegation(
+        &mut context.banks_client,
+        &context.payer,
+        &context.last_blockhash,
+    )
+    .await;
+
+    let stake_pool_accounts = StakePoolAccounts::default();
+    stake_pool_accounts
+        .initialize_stake_pool(
+            &mut context.banks_client,
+            &context.payer,
+            &context.last_blockhash,
+            MINIMUM_RESERVE_LAMPORTS + current_minimum_delegation + stake_rent,
+        )
+        .await
+        .unwrap();
+
+    let source_validator_stake = simple_add_validator_to_pool(
+        &mut context.banks_client,
+        &context.payer,
+        &context.last_blockhash,
+        &stake_pool_accounts,
+        None,
+    )
+    .await;
+
+    let destination_validator_stake = simple_add_validator_to_pool(
+        &mut context.banks_client,
+        &context.payer,
+        &context.last_blockhash,
+        &stake_pool_accounts,
+        None,
+    )
+    .await;
+
+    let minimum_redelegate_lamports = current_minimum_delegation + stake_rent * 2;
+    simple_deposit_stake(
+        &mut context.banks_client,
+        &context.payer,
+        &context.last_blockhash,
+        &stake_pool_accounts,
+        &source_validator_stake,
+        minimum_redelegate_lamports,
+    )
+    .await
+    .unwrap();
+
+    let mut slot = 0;
+    if do_warp {
+        slot = context.genesis_config().epoch_schedule.first_normal_slot;
+        context.warp_to_slot(slot).unwrap();
+        stake_pool_accounts
+            .update_all(
+                &mut context.banks_client,
+                &context.payer,
+                &context.last_blockhash,
+                &[
+                    source_validator_stake.vote.pubkey(),
+                    destination_validator_stake.vote.pubkey(),
+                ],
+                false,
+            )
+            .await;
+    }
+
+    let last_blockhash = context
+        .banks_client
+        .get_new_latest_blockhash(&context.last_blockhash)
+        .await
+        .unwrap();
+
+    (
+        context,
+        last_blockhash,
+        stake_pool_accounts,
+        source_validator_stake,
+        destination_validator_stake,
+        minimum_redelegate_lamports,
+        slot,
+    )
+}
+
+#[tokio::test]
+async fn success() {
```

Comment:
it would be nice if the test names were prefixed with `redelegate_` so that `cargo test-sbf redelegate` does this expected thing

Reply:
ah, I Just do `cargo test-sbf --test redelegate` to scope the runs. Since you can scope that way, I think it's redundant to add `redelegate_` again in the test name. Is that ok?

Reply:
ooh til